sentry/src/main/java/io/sentry/Span.java

```diff
@@ -334,13 +336,34 @@ public void setData(@NotNull String key, @NotNull Object value) {
 
   @Override
   public void setMeasurement(@NotNull String name, @NotNull Number value) {
-    this.transaction.setMeasurement(name, value);
+    if (isFinished()) {
+      return;
+    }
+    this.measurements.put(name, new MeasurementValue(value, null));
+    // We set the measurement in the transaction, too, but we have to check if this is the root span
+    // of the transaction, to avoid an infinite recursion
+    if (transaction.getRoot() != this) {
+      transaction.setMeasurement(name, value);
```

Comment:
this is already happening, right?
we may add a check in the `SentryTracer` setMeasurement if a measurement already exists. Wdyt?

Reply:
looks like it would be a [breaking change](https://github.com/getsentry/sentry-java/blob/feat/span-level-measurement/sentry/src/test/java/io/sentry/SentryTracerTest.kt#L972-L987), technically
@romtsn some input here, too? 😄

Reply:
I guess I'd be expecting the last value to be preserved when setting same measurement multiple times. Should we maybe have an interal `setMeasurement(name, value, fromChild)` method for this specific case in SentryTracer? i.e. when we set it from child spans we don't overwrite, but when called directly on the transaction then we do overwrite, how does that sound?